sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -662,7 +662,7 @@ func (c *control) handleInstruction(ctx context.Context, req *fnpb.InstructionRe
 				},
 			},
 		}
-	case req.GetSampleData() != nil:
+	case req.GetSampleData() != nil && c.dataSampler != nil:
```

Comment:
This isn't an ideal fix. 

As much as the Runner shouldn't be sending sample requests, this sends "fail" response back to the Runner, which may abort job processing depending on the runner.

It would be better to send "empty" responses with no sampleData in this case instead. 
```
if c.dataSampler != nil {
	return &fnpb.InstructionResponse{
			InstructionId: string(instID),
			Response: &fnpb.InstructionResponse_SampleData{},
		}
}
```

Reply:
Basically, yes, Dataflow is incorrect to be sending these requests, however, the SDK shouldn't send responses that might cause the job to fail.

Reply:
Updated with empty response when data sampling is not enabled.